File: Definitions/Game_Definitions/classes/GiveRewardEffector.lua
---@meta
---@diagnostic disable

---@class GiveRewardEffector : gameEffector
---@field reward TweakDBID
---@field target entEntityID
GiveRewardEffector = {}

---@return GiveRewardEffector
function GiveRewardEffector.new() return end

---@param props table
---@return GiveRewardEffector
function GiveRewardEffector.new(props) return end

---@param owner gameObject
function GiveRewardEffector:ActionOff(owner) return end

---@param owner gameObject
function GiveRewardEffector:ActionOn(owner) return end

---@param record TweakDBID|string
---@param parentRecord TweakDBID|string
function GiveRewardEffector:Initialize(record, parentRecord) return end

---@param owner gameObject
function GiveRewardEffector:ProcessAction(owner) return end

---@param owner gameObject
function GiveRewardEffector:RepeatedAction(owner) return end

function GiveRewardEffector:Uninitialize() return end

File: DecompiledGameScripts/core/gameplay/effectors/singulars/giveRewardEffector.swift

public class GiveRewardEffector extends Effector {

  public let m_reward: TweakDBID;

  public let m_target: EntityID;

  protected func Initialize(record: TweakDBID, game: GameInstance, parentRecord: TweakDBID) -> Void {
    let str: String = TweakDBInterface.GetString(record + t".reward", "");
    this.m_reward = TDBID.Create(str);
  }

  protected func Uninitialize(game: GameInstance) -> Void;

  private final func ProcessAction(owner: ref<GameObject>) -> Void {
    this.GetApplicationTarget(owner, n"Target", this.m_target);
    if EntityID.IsDefined(this.m_target) {
      RPGManager.GiveReward(owner.GetGame(), this.m_reward, Cast<StatsObjectID>(this.m_target));
    } else {
      RPGManager.GiveReward(owner.GetGame(), this.m_reward);
    };
  }

  protected func RepeatedAction(owner: ref<GameObject>) -> Void {
    this.ProcessAction(owner);
  }

  protected func ActionOn(owner: ref<GameObject>) -> Void {
    this.ProcessAction(owner);
  }

  protected func ActionOff(owner: ref<GameObject>) -> Void;
}
